src/DeviceInterface/InterfacePrototype.js

```diff
@@ -254,16 +321,121 @@ class InterfacePrototype {
         if (isMobileApp) {
             this.getAlias().then((alias) => {
                 if (alias) form.autofillEmail(alias)
-                else form.activeInput.focus()
+                else form.activeInput?.focus()
             })
-        } else {
-            this.attachTooltipInner(form, input, inputType, getPosition, click)
+            return
         }
+
+        /** @type {TopContextData} */
+        const topContextData = {
+            inputType
+        }
+
+        // A list of checks to determine if we need to generate a password
+        const checks = [
+            inputType === 'credentials.password',
+            this.supportsFeature('password.generation'),
+            form.isSignup
+        ]
+
+        // if all checks pass, generate and save a password
+        if (checks.every(Boolean)) {
+            const password = this.createAndStorePassword(form.form, input)
+
+            // append the new credential to the topContextData so that the top autofill can display it
+            topContextData.credentials = [fromPassword(password)]
+        }
+
+        this.attachTooltipInner(form, input, getPosition, click, topContextData)
     }
 
-    attachTooltipInner (form, input, inputType, getPosition, _click) {
+    /**
+     * Create a new password once for a given HTMLFormElement
+     *
+     * @param {HTMLFormElement} formElement
+     * @param {HTMLInputElement} input
+     * @returns {string}
+     */
+    createAndStorePassword (formElement, input) {
```

Comment:
@GioSensation yes @jonathanKingston has also suggested generating the Password once per page also - if we believe this is a safer option it would be fairly straight forward to implement, but as you say, possibly a follow-on?